src/main/kotlin/com/sourcegraph/cody/edit/DocumentCodeSession.kt

```diff
@@ -0,0 +1,33 @@
+package com.sourcegraph.cody.edit
+
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Editor
+import com.sourcegraph.cody.agent.CodyAgent
+import com.sourcegraph.cody.agent.protocol.EditTask
+import java.util.concurrent.CompletableFuture
+
+class DocumentCodeSession(controller: FixupService, editor: Editor) :
+    FixupSession(controller, editor) {
+  private val logger = Logger.getInstance(DocumentCodeSession::class.java)
+
+  override fun makeEditingRequest(agent: CodyAgent): CompletableFuture<EditTask> {
+    return agent.server.commandsDocument()
+  }
+
+  override fun retry() {
+    // TODO: The actual prompt is displayed as ghost text in the text input field.
+    // E.g. "Write a brief documentation comment for the selected code <etc.>"
+    // We need to send the prompt along with the lenses, so that the client can display it.
+    EditCommandPrompt(controller, editor, "Edit instructions and Retry").displayPromptUI()
+  }
+
+  // Brings up a diff view showing the changes the AI made.
+  override fun diff() {
+    // The FixupController issues a vscode.diff command to show the smart diff in the
```

Comment:
No longer true. The VSCode code lens UI does this, but we're using the "client UI" now and I made the vscode.diff an implementation detail of the VSCode code lens UI. (Of course, it is software, we can do whatever we want. But that's where the Agent/extension/VSCode side changes are at right now.)

Reply:
fixed